### ISSUE TITLE
[BUG] - Fedora CoreOS Ignition file doesn't work from terraform

### ISSUE THREAD

**Issue by nsmith5** [vultr/terraform-provider-vultr#68]
If I use `user_data` to supply an Ignition file for Fedora CoreOS the system doesn't boot (presumably because the ignition file is malformed). If I supply the same file in the WebUI it works and if I supply the same file to a local qemu VM it also works. I can't seem to get this to work regardless of the ignition file.

**To Reproduce**
Create a `vultr_instance` that references an ignition file

```hcl
# Example
resource "vultr_server" "example" {
  plan_id = "401"
  region_id = "4"

  # Fedora CoreOS
  os_id = "391"
  
  user_data = file("/path/to/server.ign")
}
```
On `terraform apply` you'll find that your the server doesn't boot.

**Expected behavior**
Server should boot as the ignition file is has been validated an many other places

**Screenshots**

**Desktop (please complete the following information where applicable:**
 - Terraform 0.13.5
 - Local machine: Fedora Workstation 32
 - Server machine: Fedora CoreOS 32
 - Terraform provider version: 1.4.1

**Additional context**
I understand the userdata is base64 encoded and sent to the API. Should it be encoded in a URL safe fashion? I see its not right now? Otherwise I can't figure out the problem.

**Comment by nsmith5** [comment]
I managed to get an emergency console up to debug and see the following:

![Screenshot from 2020-10-30 18-12-53](https://user-images.githubusercontent.com/12156185/97767808-dba33b80-1adb-11eb-991f-3f77152365f8.png)

As you can image... I scratched my head a bit about the first character being 'e'. Makes me think the user data is being base64 encoded. I verified this by base64 encoding my ignition file and checking if it matched the sha512 check sum and it did.

```shell
$ base64 -w 0 server.ign
ewo......      # <- begins with 'e'
$  base64 -w 0 server.ign | sha512sum  # Matches the sha512!
666dedb2e91262cb6477c8f5ec547408cc2e19766e5d3a18b06ff1bb8d91bc4ec6f7525e17ad8f8603be5d52febbdf7e0ccf733bdf5ed38cab34671a512e60ba  -
```
Maybe a mistake base64 encoding twice? Or perhaps the API isn't decoding for some reason?

**Comment by nsmith5** [comment]
base64 encoding twice seems to be happening if I look 

- [here in the terraform provider](https://github.com/vultr/terraform-provider-vultr/blob/09a16af1ef036bc7ecc7b161bf148aad88861ea9/vultr/resource_vultr_server.go#L263)
- [and then here in govultr](https://github.com/vultr/govultr/blob/83ab8ff053b3fdd9afc3d85ba186a2dd39d2fb16/server.go#L1352)

perhaps we don't need the addition base64 encoding in terraform provider?

**Comment by ddymko** [comment]
@nsmith5 thanks for looking into this.

With API V1 we made a change where it expects `userdata` to already be b64 encoded and it will reject the call if it is not b64 encoded.

We did update GoVultr to no longer b64 user data and let the API itself reject the call if it wasn't b64'd. However it looks like we missed making this change for create https://github.com/vultr/govultr/pull/73. So we are double encoding here for create.

We intentionally left Terraform to take in strings and we would encode it for you to not introduce breaking changes. 

So here are the action items we'll take:

- Double check the API to make sure it isn't doing anything funky with the userdata and that its enforcing the b64 on userdata.
- Update GoVultr V1 to no longer encode b64 on create (as it is on update) for instance and baremetal.
- As you pointed out in https://github.com/vultr/terraform-provider-vultr/issues/69 we'll make this change as well it didn't cross our minds until you pointed it out
